core/slim_sim.cpp

```diff
@@ -8101,48 +8246,42 @@ slim_generation_t SLiMSim::_InitializePopulationFromTskitTextFile(const char *p_
 
 slim_generation_t SLiMSim::_InitializePopulationFromTskitBinaryFile(const char *p_file, EidosInterpreter *p_interpreter)
 {
+    int ret;
+
 	if (!recording_tree_)
 		EIDOS_TERMINATION << "ERROR (SLiMSim::_InitializePopulationFromTskitBinaryFile): to load a tree-sequence file, tree-sequence recording must be enabled with initializeTreeSeq()." << EidosTerminate();
 	
 	// free the existing table collection
 	FreeTreeSequence();
 	
-#if 0
-	// CRASHES: the loaded table collection is immutable (non-malloc'd columns)
-	// Jerome says this is a bug, and that he will add a flag for tsk_table_collection_load() to make it copy the blocks
-	// read the file from disk
-	ret = tsk_table_collection_load(&tables, p_file, 0);
-	if (ret != 0) handle_error("tsk_table_collection_load", ret);
-#else
-	// WORKAROUND
-	// read the file from disk into a private table collection that is immutable
-	tsk_table_collection_t immutable_tables;
-	
-	int ret = tsk_table_collection_load(&immutable_tables, p_file, 0);
+	ret = tsk_table_collection_load(&tables_, p_file, 0);
 	if (ret != 0) handle_error("tsk_table_collection_load", ret);
 	
-	// BCH 4/25/2019: if indexes are present on immutable_tables we want to drop them; they are synced up
+	// BCH 4/25/2019: if indexes are present on tables_ we want to drop them; they are synced up
 	// with the edge table, but we plan to modify the edge table so they will become invalid anyway, and
-	// then they can cause a crash because of their unsynced-ness; see tskit issue #179
-	ret = tsk_table_collection_drop_index(&immutable_tables, 0);
+	// then they may cause a crash because of their unsynced-ness; see tskit issue #179
+	ret = tsk_table_collection_drop_index(&tables_, 0);
 	if (ret != 0) handle_error("tsk_table_collection_drop_index", ret);
-	
-	// copy the immutable table collection to make a mutable collection
-	ret = tsk_table_collection_copy(&immutable_tables, &tables_, 0);
-	if (ret < 0) handle_error("tsk_table_collection_copy", ret);
 
 	RecordTablePosition();
 	
 	// convert ASCII derived-state data, which is the required format on disk, back to our in-memory binary format
 	DerivedStatesFromAscii(&tables_);
 	
-	// in nucleotide-based models, read the ancestral sequence from the open kastore of immutable_tables
+	// in nucleotide-based models, read the ancestral sequence
 	if (nucleotide_based_)
 	{
 		char *buffer;				// kastore needs to provide us with a memory location from which to read the data
 		std::size_t buffer_length;	// kastore needs to provide us with the length, in bytes, of the buffer
+        kastore_t *store;
+
+        ret = kastore_open(store, p_file, "r", 0);
```

Comment:
I think there's a bug here...?  As far as I can tell, `kastore_open()` expects to be passed a pointer to a `kastore_t` record that it then zeros and initializes.  Here, you're instead passing an uninitialized pointer to nowhere; so `kastore_open()` will zero and initialize some random chunk of memory somewhere.  I'm surprised this didn't crash any of your tests; maybe none of them are nucleotide-based, so this code path doesn't get hit?  Anyway, you want to do something like:

```
kastore_t store;

ret = kastore_open(&store, p_file, "r", 0);
...
kastore_close(&store);
```

Anyway, that's how it seems to me from looking at the kastore code.

Reply:
Huh. My tests *are* using nucleotide sequences, and getting the reference sequence OK.   But, you're right! Whoops!

Reply:
Just lucky it doesn't crash, then.  Probably sometimes it would.